client/state/data-layer/extensions-middleware.js

```diff
@@ -1,101 +1,58 @@
 /**
- * External dependencies
- */
-import { toPairs, fromPairs } from 'lodash';
-
-/**
  * Internal dependencies
  */
-import { local, mergeHandlers } from './utils';
-
-const stripHandlersForKey = ( key, keyHandlers, keyHandlersToStrip ) => {
-	const newKeyHandlers = (
-		keyHandlersToStrip
-			? keyHandlers.filter( ( handler ) => ! keyHandlersToStrip.includes( handler ) )
-			: keyHandlers
+import { mergeHandlers } from './utils';
+import { middleware } from './wpcom-api-middleware';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:state:data-layer' );
+let extensionHandlers = Object.create( null );
+let handlerMiddleware = buildMiddleware( extensionHandlers );
+
+export function addHandlers( extensionName, handlers, existingHandlers = extensionHandlers ) {
+	if ( Object.keys( existingHandlers ).includes( extensionName ) ) {
```

Comment:
warning! I don't think this will work in IE - `Array.prototype.includes`

it would be satisfactory though to perform a simpler check without extracting all of the keys…

```js
if ( undefined !== existingHandlers[ extensionName ] )

// or, maybe more semantically, maybe insignificantly less performant

if ( existingHandlers.hasOwnProperty( extensionName ) )
```